OfflineService/OfflineServiceViewController.swift

```diff
@@ -0,0 +1,311 @@
+    }
+    
+    // MARK: - Action handler methods
+    
+    @IBAction func dismissViewController() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    // MARK: - UITableView delegate methods
+    
+    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath) as! OfflineDataRegionTableViewCell
+        cell.presentUI(for: offlineDataItems[indexPath.row])
+
+        return cell
+    }
+    
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return 1
+    }
+    
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return offlineDataItems.count
+    }
+    
+    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 200.0
+    }
+    
+    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        tableView.deselectRow(at: indexPath, animated: true)
+        
+        let offlineDataRegion = offlineDataItems[indexPath.row]
+        presentActionsAlertController(offlineDataRegion)
+    }
+    
+    func listAvailableRegions() {
+        guard let outputDirectory = Bundle.mapboxCoreNavigation.suggestedTileURL?.path else { return }
+        if !Bundle.mapboxCoreNavigation.ensureSuggestedTileURLExists() {
+            print("Failed to create tiles directory")
+            return
+        }
+        
+        removeUnpackedTilesDirectory()
+        
+        offlineService = OfflineService.getInstanceForPath(outputDirectory, options: OfflineServiceOptions(username: username,
+                                                                                                           accessToken: accessToken,
+                                                                                                           baseURL: baseURL))
+
+        offlineService?.listAvailableRegions(forCallback: { [weak self] (expected) in
+            guard let self = self else { return }
+            if let error = expected?.error as? OfflineDataError {
+                DispatchQueue.main.async {
+                    self.presentAlert(OfflineServiceConstants.title, message: error.message)
+                }
+
+                return
+            }
+
+            let offlineDataRegions = expected?.value as? Array<Any>
+            offlineDataRegions?.forEach({
+                if let metadata = $0 as? OfflineDataRegionMetadata {
+                    self.offlineDataItems.append(OfflineDataItem(dataRegionMetadata: metadata,
+                                                                 mapPackMetadata: nil,
+                                                                 navigationPackMetadata: nil))
+                }
+            })
+
+            OfflineService.registerObserver(for: self)
```

Comment:
I think we should register `self` on view did load

Reply:
Moved to `OfflineServiceViewController.listAvailableRegions()` for now.